plugins/nsd/nsd

```diff
@@ -0,0 +1,127 @@
+#!/bin/sh
+
+: << =cut
+
+=head1 NAME
+
+nsd - Plugin to monitor nsd DNS server
+
+=head1 CONFIGURATION
+
+No configuration
+
+=head1 AUTHOR
+
+Kim Heino <b@bbbs.net>
+
+=head1 LICENSE
+
+GPLv2
+
+=head1 MAGIC MARKERS
+
+ #%# family=auto
+ #%# capabilities=autoconf
+
+=cut
+
+if [ "$1" = "autoconf" ]; then
+    if [ -x /usr/sbin/nsd-control ]; then
+        echo "yes"
+        exit 0
+    else
+        echo "no (no /usr/sbin/nsd-control)"
+        exit 0
+    fi
+fi
+
+if [ "$1" = "config" ]; then
+    echo 'graph_title NSD queries'
+    echo 'graph_vlabel queries / second'
+    echo 'graph_category dns'
+    echo 'graph_info Queries per second, by query type'
+    echo 'a.label A'
+    echo 'a.type DERIVE'
+    echo 'a.min 0'
+    echo 'aaaa.label AAAA'
+    echo 'aaaa.type DERIVE'
+    echo 'aaaa.min 0'
+    echo 'ptr.label PTR'
+    echo 'ptr.type DERIVE'
+    echo 'ptr.min 0'
+    echo 'cname.label CNAME'
+    echo 'cname.type DERIVE'
+    echo 'cname.min 0'
+    echo 'mx.label MX'
+    echo 'mx.type DERIVE'
+    echo 'mx.min 0'
+    echo 'txt.label TXT'
+    echo 'txt.type DERIVE'
+    echo 'txt.min 0'
+    echo 'soa.label SOA'
+    echo 'soa.type DERIVE'
+    echo 'soa.min 0'
+    echo 'ns.label NS'
+    echo 'ns.type DERIVE'
+    echo 'ns.min 0'
+    echo 'srv.label SRV'
+    echo 'srv.type DERIVE'
+    echo 'srv.min 0'
+    echo 'dnskey.label DNSKEY'
+    echo 'dnskey.type DERIVE'
+    echo 'dnskey.min 0'
+    echo 'axfr.label AXFR'
+    echo 'axfr.type DERIVE'
+    echo 'axfr.min 0'
+    echo 'snxd.label NXDOMAIN'
+    echo 'snxd.type DERIVE'
+    echo 'snxd.min 0'
+    echo 'rq.label Total Successful'
+    echo 'rq.type DERIVE'
+    echo 'rq.min 0'
+    exit 0
+fi
+
+/usr/sbin/nsd-control stats_noreset | sed 's/=/ /; s/\.//g' | (
+    numtypeA=0
```

Comment:
Maybe `U` (failed to retrieve value) would be more suitable here?
Or should the missing key being interpreted as *no requests of this type*?